### PR TITLE
Migrate Config to dconf

### PR DESCRIPTION
**Pull request by cmkohnen**
Closes https://github.com/Sjoerd1993/Graphs/issues/389

This gets rid of the old logic for preferences and replaces it with calls to `Gio.Settings`.
Settings are now grouped by category.
Settings which are a selection are now an enum.
A few values have been renamed to better describe their function.

Old config files are automatically migrated and thereafter deleted.

### CODE REVIEW

**Review by Sjoerd1993 — CHANGES_REQUESTED** [review]
That was a pretty hefty refactor, thanks a lot for this work. It's looking good overall.

See comments beneath. About the enums, I'm fine with using ints, just wanted to throw the question out there. Otherwise there's some bugs. Other than the ones I pointed out, it seems to work well. Also with old data and configs.

File: data/meson.build [hunk 1]
```diff
@@ -38,15 +38,9 @@ if appstream_util.found()
...
-        input: application_id + '.gschema.xml.in',
-       output: application_id + '.gschema.xml',
-configuration: conf
-)
-
-install_data(gschema_file,
+install_data(application_id + '.gschema.xml',
   install_dir: join_paths(datadir, 'glib-2.0/schemas'),
-    install_mode: 'rwxrwxrwx'
+ install_mode: 'rwxrwxrwx'
```
[the remark this answers is not in the document]
Comment (cmkohnen) [reply data/meson.build:43]: Throughout the rest of this file it is aligned by the ":" character, but I don't have a strong preference.

File: src/graphs.py [hunk 2]
```diff
@@ -48,11 +48,12 @@ def add_items(self, items):
...
         return
     ignored = []
-    handle_duplicates = self.preferences["other_handle_duplicates"]
+    handle_duplicates = \
+        self.settings.get_child("general").get_enum("handle-duplicates")
     for item in items:
         for item_1 in self.datadict.values():
             if item.name == item_1.name:
-                if handle_duplicates == "Auto-rename duplicates":
+                if handle_duplicates == 0:  # Auto-add
```
[the remark this answers is not in the document]
Comment (cmkohnen) [reply src/graphs.py:56]: The main reason I chose this approach was, that I'm not sure if settings are translated when collected via `settings.get_string()`. Using enums was a means to ensure, this doesn't become a problem.

I don't have a strong preference here and nothing would prevent us from simply getting `get_string()` and do a string comparison here (although it is an enum). I don't know if Gio handles this automatically, but removing the second option would definetely be migratable.

File: src/misc.py [hunk 3]
```diff
@@ -48,22 +52,6 @@ def _(message):
...
-]
-X_POSITIONS = [_("top"), _("bottom")]
-Y_POSITIONS = [_("left"), _("right")]
-SEPARATORS = [",", "."]
-ACTION_CENTER_DATA = [
-    _("Center at maximum Y value"), _("Center at middle coordinate"),
-]
-HANDLE_DUPLICATES = [
-    _("Auto-rename duplicates"), _("Ignore duplicates"), _("Add duplicates"),
-    _("Override existing items"),
```
[the remark this answers is not in the document]
Comment (cmkohnen) [reply src/misc.py:65]: Oh, I didn't notice these wer used there as well, thanks for noticing

File: src/export_figure.py [hunk 4]
```diff
@@ -3,73 +3,56 @@
...
 
     @Gtk.Template.Callback()
     def on_accept(self, _button):
-        dpi = int(self.dpi.get_value())
-        fmt = utilities.get_selected_chooser_item(self.file_format)
-        file_suffixes = None
-        for name, formats in self.items:
-            if name == fmt:
-                file_suffixes = formats
+        file_format = self.file_format.get_selected_item()
```
[the remark this answers is not in the document]
Comment (cmkohnen) [reply src/export_figure.py:34]: Slipped past me, thanks for noticing

File: src/ui.py [hunk 5]
```diff
@@ -185,3 +185,40 @@ def save_values_to_dict(window, keys: list):
...
+                widget.connect(
+                    "notify::selected", _on_settings_select, settings, key)
+            elif isinstance(widget, Gtk.SpinButton):
+                settings.bind(key, widget, "value", 0)
+            elif isinstance(widget, Gtk.Switch):
+                settings.bind(key, widget, "active", 0)
+            elif isinstance(widget, Adw.ExpanderRow):
+                settings.bind(key, widget, "enable-expansion", 0)
+            elif isinstance(widget, Gtk.Scale):
+                settings.bind(key, widget, "value", 0)
```
[the remark this answers is not in the document]
Comment (cmkohnen) [reply src/ui.py:220]: A lot simpler, than what I would've done, ty

**Review by Sjoerd1993 — APPROVED** [review]

**Pull request merged**